Make Blank Canvas Blocks

pbking opened the pull request:
Blank Canvas is an unopinionated parent theme that will "Ponyfill" the
gaps between where Gutenberg is today and where it is going.

The Milestone for the creation of Blank Canvas is here: https://github.com/Automattic/themes/milestone/140 

This PR should be considered ready to merge when that milestone has been achieved and any changes have been reviewed and merged into this branch.

pbking committed:
First Build of Blank Canvas Blocks

Blank Canvas is an unopinionated parent theme that will "Ponyfill" the
gaps between where Gutenberg is today and where it is going.

refactored (ponyville) to blank canvas blocks

MaggieCabrera reviewed (approved):
LGTM, we are holding off on merging this though

MaggieCabrera commented:
Closes #3412

jffng commented on blank-canvas-blocks/sass/blocks/_style.scss:
```diff
@@ -0,0 +1,4 @@
+@import "button/style";
```
Looks like this hasn't been brought over yet so this line needs to be removed or the button styles added.

```suggestion
```

jffng commented on blank-canvas-blocks/functions.php:
```diff
@@ -0,0 +1,30 @@
+
+		// Add support for editor styles.
+		add_theme_support( 'editor-styles' );
+
```
If editor styles are supported, do we also need to `add_theme_support( 'wp-block-styles' )` to avoid mismatches between the front-end and view?

```suggestion
		// Add support for editor styles.
		add_theme_support( 'wp-block-styles' );
```

jffng committed:
Match quote citation font size.

jffng commented on blank-canvas-blocks/experimental-theme.json:
```diff
@@ -0,0 +1,267 @@
+					},
+					{
+						"slug": "blue",
+						"color": "007cba",
```
```suggestion
						"color": "#007cba",
```

jffng committed:
Update border and font family styles.

jffng committed:
Add padding.

jffng committed:
Rename to primary and secondary.

pbking committed:
Adding the missing 'button' styles

jffng committed:
Configure to only use single base font.

jffng committed:
Remove font families from quote styles.

pbking committed:
Adjusted button colors

pbking committed:
Changes to deploy action

pbking committed:
Blank Canvas Blocks: Add alignment rules (#3429)

* Add alignment rules for Blank Canvas Blocks

* Adjusted configured with to match original BC theme

Co-authored-by: Ben Dwyer <ben@scruffian.com>

pbking committed:
Getting button styles re-introduced to BCB (#3427)

pbking committed:
Added support for 'reasonable default' block styles

pbking committed:
Merge branch 'make/blank-canvas-blocks' of github.com:Automattic/themes into make/blank-canvas-blocks

pbking committed:
Fixing deployment destination target

MaggieCabrera committed:
refactored css folder structure (#3431)

jffng committed:
Add default styles for Separator block (#3432)

* Implement custom separator variables.

pbking committed:
Adjusting font size configuration to match Blank Canvas (#3433)

MaggieCabrera committed:
Blan Canvas Blocks: Add default styles for Video block (#3434)

* add styles for video block, responsive embeds support, expressed caption styles via theme.json

danieldudzic committed:
Blank Canvas Blocks: Add default styles for List block

MaggieCabrera committed:
switch to camel case

scruffian committed:
Merge pull request #3444 from Automattic/bcb-camel-case

Blank Canvas Blocks: Switch custom variables to camel case

MaggieCabrera committed:
added gallery styles

MaggieCabrera committed:
font size rule fixed

scruffian committed:
Blank Canvas Blocks: Add support for Google fonts

pbking committed:
Add default styles to BCB for Image Block (#3430)

* Added default styles for image blocks.

* Refactored margins to simplify

* Rearranged block sass imports to match source branch

* rebuilt with latest changes

* Fix image, video and audio block alignments

* Added Alignmnent styles for wide/full for editor; Tightened up image styles (also for editor)

Co-authored-by: Ben Dwyer <ben@scruffian.com>

pbking committed:
Add configurable header styles to Blank Canvas Blocks (#3439)

Added configurable Header settings to BCB

pbking committed:
Add paragraph drop cap style configuration (#3436)

* Add paragraph drop cap style configuration

Co-authored-by: Jeff Ong <jonger4@gmail.com>

jffng committed:
Add image reset styles.

danieldudzic committed:
Blank Canvas Blocks: Fix List block styles

danieldudzic committed:
Blank Canvas Blocks: Fix List block font-family ponyfill variable

scruffian committed:
Merge pull request #3437 from Automattic/add/bcb/list-block-styles

Blank Canvas Blocks: Add default styles for List block

MaggieCabrera committed:
Merge pull request #3450 from Automattic/add/bcb-img-reset

Add image resets

MaggieCabrera committed:
removed unnecesary styles

MaggieCabrera committed:
more css pruning

MaggieCabrera committed:
fix from rebase

MaggieCabrera committed:
Merge pull request #3446 from Automattic/bcb-gallery-block

Blank Canvas Blocks: Add default styles for Gallery block

MaggieCabrera committed:
reordered sass includes, created search block css file

scruffian committed:
Update to use the CSS2 Google API

pbking committed:
Adjusted Group Alignment especially for nested groups (#3449)

* Adjusted Group Alignment especially for nested groups

Co-authored-by: Jeff Ong <jonger4@gmail.com>

jffng committed:
Rebase make/bcb.

jffng committed:
Fix font family URL.

jffng committed:
Remove accidental check in.

scruffian committed:
Load Google Fonts from theme.json

pbking committed:
Refactored to simplify base alignment styles to better target incoming Gutenberg changes.

pbking committed:
removed 'default vertical padding' from all the elements

scruffian committed:
remove testing code

scruffian committed:
fix a warning in the case where the fonts aren't defined

scruffian committed:
Merge pull request #3447 from Automattic/add/bcb-fonts

Blank Canvas Blocks: Add Google fonts

MaggieCabrera committed:
search block styles started

scruffian committed:
Restore left/right/centre alignments

scruffian committed:
Merge pull request #3459 from Automattic/refactor/bcb-alignmnet

Refactor/bcb alignmnet

MaggieCabrera committed:
more work on search block css

MaggieCabrera committed:
comment wording changed

MaggieCabrera committed:
changed margin to button instead of input

scruffian committed:
Blank Canvas Blocks: Remove vertical margins from blocks in the editor (#3471)

* Editor: Removes the vertical margins from blocks in the editor

pbking committed:
Add Navigation styling to BCB (#3460)

* Add navigation style configuration to theme.json & add drop-shadow to submenu as an option.

* Adding mobile menu option to target mobile navigation styles exclusively for navigation blocks in the header.

scruffian committed:
Blank Canvas Blocks: Enqueue editor styles on admin_init

scruffian committed:
Remove testing changes

scruffian committed:
Merge pull request #3482 from Automattic/fix/getting-data-from-theme-json

Blank Canvas Blocks: Fix getting data from theme json

pbking committed:
Refactor links in BCB to use underline instead of border-bottom. (#3478)

* Refactored link elements to use underline instead of border-bottom
* Refactored away not selectors from link styles.
Co-authored-by: Jeff Ong <jeff.ong@automattic.com>
Co-authored-by: Ben Dwyer <ben@scruffian.com>

pbking committed:
Refactor/bcb custom to styles (#3487)

Refactored "custom" style configuration into "style" configuration for blocks wherever possible. 

This was enabled because Gutenberg recently enabled configuration of styles of blocks even if they don't express support via their block.json.

This also brings alignment configuration in line with recent changes allowing blocks to express "wide/full" configuration individually.

pbking committed:
Refactored Search block to minimize overrides and pull details from form.

pbking committed:
Removed unused custom settings and comments regarding them.

scruffian committed:
BCB: Don't load block styles in the editor (#3495)

pbking committed:
Merge branch 'make/blank-canvas-blocks' into bcb-search-block

pbking committed:
Introduce default Button Block padding to non-block buttons

scruffian committed:
Merge pull request #3469 from Automattic/bcb-search-block

Blank Canvas Blocks: Add default styles for search block

pbking committed:
Style BCB File Block Buttons (#3494)

* Added button styles to file block

Co-authored-by: Ben Dwyer <ben@scruffian.com>

scruffian committed:
Use variables from theme.json for table styles

pbking committed:
Added padding to code block styles

scruffian committed:
Move table CSS to sub object

scruffian committed:
fix alphabetization

jffng committed:
Merge pull request #3503 from Automattic/add/table-styles

Blank Canvas Blocks: Add table styles

jffng committed:
Merge pull request #3506 from Automattic/add/bcb-code-padding

Added padding to code block styles

scruffian committed:
Blank Canvas Blocks: Add Pullquote Block styles (#3501)

* Add rules for pullquotes

* Refactored pullquote and quote to use Gutenberg styles instead of custom where possible

Co-authored-by: Jason Crist <jcrist@pbking.com>

scruffian committed:
Blank Canvas Blocks: Use Gutenberg breakpoints. (#3505)

* Blank Canvas Blocks: Replace our custom breakpoints with the gutenberg defaults

pbking committed:
Limit content width for things aligned left/right (#3504)

* Limit content width for things aligned left/right

Co-authored-by: Jeff Ong <jeff.ong@automattic.com>

pbking committed:
Fixed media query error from previous merge

pbking committed:
Refactored lef/right alighnment reset

MaggieCabrera committed:
Blank Canvas Blocks: Added Block Patterns (#3514)

* added block patterns

* Support editor color assignment for separator.
* Refactor separator styles.

Co-authored-by: Ben Dwyer <ben@scruffian.com>
Co-authored-by: Jason Crist <jcrist@pbking.com>
Co-authored-by: Jeff Ong <jonger4@gmail.com>

danieldudzic committed:
Blank Canvas Blocks: Update templateParts json (#3528)

scruffian committed:
Blank Canvas Blocks: Add a build tool

scruffian committed:
use a js file instead of a php one

pbking committed:
Work on updating the default templates to better match Blank Canvas (#3522)

* Update the default templates to better match Blank Canvas

* Added footer template part definition

scruffian committed:
add missing file

scruffian committed:
Merge pull request #3536 from Automattic/add/build-tool-for-bcb

Blank Canvas Blocks: Add a build tool

pbking committed:
Format generated JSON (#3548)

danieldudzic committed:
Blank Canvas Blocks: Remove hover states for buttons with custom background colors

danieldudzic committed:
Blank Canvas Blocks: Re-add svg button styling

scruffian committed:
Merge pull request #3556 from Automattic/fix/bcb-button-hover

Blank Canvas Blocks: Remove hover states for buttons with custom background colors

scruffian committed:
Blank Canvas Blocks: Update build script

scruffian committed:
Merge pull request #3562 from Automattic/update/bcb-build-script

Blank Canvas Blocks: Update build script

pbking committed:
Updated metadata to (mostly) match Blank Canvas

pbking commented:
c-D59498
